content/documentation/4-batch-developer-guides/2-batch/2-spring-batch.md

```diff
@@ -8,34 +8,310 @@ description: 'Create a Spring Batch Job'
+
+For this section we will create a Spring Cloud Task/Spring Batch bill run application that will read usage information from a json file containing customer usage data and price each entry and place the result into the `BILL_STATEMENTS` table.
+
+![BILL_STATMENTS](images/bill_statements.png)
 
-Create a single step batch job, file item reader, file item writer, no other dependencies outside of spring batch.
+This will be done using a single step batch job, `JsonItemReader`, `BillProcessor`, `JdbcBatchItemWriter`.
 
 ### Initialzr
 
-Go to initializr - maybe have a curl command ready to go do we can just download a ready made .zip.
+1. Visit the [Spring Initialzr site](https://start.spring.io/).
+1. Select the latest release of spring boot.
+1. Create a new Maven project with a Group name of `io.spring` and an Artifact name of `billrun`.
+1. In the Dependencies text box, type `task` to select the Cloud Task dependency.
+1. In the Dependencies text box, type `jdbc` then select the JDBC dependency.
+1. In the Dependencies text box, type `h2` then select the H2 dependency.
+   1. We use H2 for unit testing.
+1. In the Dependencies text box, type `mysql` then select mysql dependency (or your favorite database).
+   1. We use MySql for the runtime database.
+1. In the Dependencies text box, type Batch then select Batch.
+1. Click the Generate Project button.
+1. Unzip the billrun.zip file and import the project into your favorite IDE.
+
+Another option instead of using the UI to initialize your project you can do the following:
+
+1. Click the [here](https://start.spring.io/starter.zip?fakeusernameremembered=&fakepasswordremembered=&type=maven-project&language=java&baseDir=billrun&groupId=io.spring&artifactId=billrun&name=Bill+Run&description=Bill+Run+Sample+App&packageName=io.spring.billrun&packaging=jar&inputSearch=&style=batch&style=cloud-task&style=jdbc&style=h2&style=mysql) to download the preconfigured billrun.zip.
+
+2. Unzip the billrun.zip file and import the project into your favorite IDE
+
+### Setting up MySql
+
+1. If you don't have an instance of MySql available to you, you can follow these instructions to run a MySql docker image for this example.
+
+   1. Pull the MySql docker image
+
+      ```bash
+      $ docker pull mysql:5.7.25
+      ```
+
+   2. Start the MySql
+
+      ```bash
+      $ docker run -p 3306:3306 --name some-mysql -e MYSQL_ROOT_PASSWORD=password  -e MYSQL_DATABASE=task -d mysql:5.7.25
+      ```
 
 ### Biz Logic
 
-Your biz logic is coded in this section,
+1.  Using your IDE create a usageinfo.json file in the resources directory.
+
+    1. Copy the data from [here](https://github.com/spring-cloud/spring-cloud-dataflow-samples/tree/master/dataflow-website/batch-developer-guides/batch/batchsamples/billrun/src/main/resources/usageinfo.json) and insert it into the usageinfo.json file you just created.
```

Comment:
github does not directly allow for the download of files.   Damien will look into it.  We can create an issue for now.

Reply:
and yes it is supposed to be /src/main/resources.   Updated.

Reply:
You can download individual files from Github by using a link constructed as follows: https://raw.githubusercontent.com/user/repository/branch/filename

More detail here: https://stackoverflow.com/questions/4604663/download-single-files-from-github

Whether you _should_ do that is another matter.

Reply:
We did try that.  And it failed to download the app.  Rather it would open another tab with the content in it.   :-(

Reply:
I suppose one would have to right click the link and choose to save it. I'd hate to put that in a set of instructions, though. (I know, because I've done it, and I hated it then, too.)

Reply:
Yeah.   In that case copy/paste is more consistent to what we're doing.